pkg/configtx/application_test.go

```diff
@@ -567,19 +567,36 @@ func TestAnchorPeer(t *testing.T) {
 		ChannelGroup: channelGroup,
 	}
 
-	expectedAnchorPeer := Address{Host: "host1", Port: 123}
 	c := ConfigTx{
 		original: config,
 		updated:  config,
 	}
 
+	anchorPeers, err := c.AnchorPeers("Org1")
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(anchorPeers).To(BeNil())
+	gt.Expect(anchorPeers).To(HaveLen(0))
+
+	expectedAnchorPeer := Address{Host: "host1", Port: 123}
 	err = c.AddAnchorPeer("Org1", expectedAnchorPeer)
 	gt.Expect(err).NotTo(HaveOccurred())
 
-	anchorPeers, err := c.AnchorPeers("Org1")
+	anchorPeers, err = c.AnchorPeers("Org1")
 	gt.Expect(err).NotTo(HaveOccurred())
-	gt.Expect(len(anchorPeers)).To(Equal(1))
+	gt.Expect(anchorPeers).To(HaveLen(1))
 	gt.Expect(anchorPeers[0]).To(Equal(expectedAnchorPeer))
```

Comment:
I'm a little hesitant with the basis for this test since it revolves around the fact that we use the same config for both the original and updated on line 570. In a real world scenario we could never actually do this since adding/removing an anchor peer only modifies the updated whereas the retriever only uses the original.

Reply:
I was working on changing just that (in what will now be a follow-on commit/PR) when I remembered that we wanted to update the functionality of this retrieval function, hence this JIRA and PR. :) I'll have that PR out later today.